extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -490,49 +477,88 @@ public String getSubChannelAvatarUrl() {
     public String getDashMpdUrl() throws ParsingException {
         assertPageFetched();
 
-        if (desktopStreamingData != null) {
-            return desktopStreamingData.getString("dashManifestUrl");
-        } else if (mobileStreamingData != null) {
-            return mobileStreamingData.getString("dashManifestUrl");
-        } else {
-            return EMPTY_STRING;
+        if (dashUrl == null) {
+            if (desktopStreamingData != null) {
+                dashUrl = desktopStreamingData.getString("dashManifestUrl", EMPTY_STRING);
+            } else if (mobileStreamingData != null) {
+                dashUrl = mobileStreamingData.getString("dashManifestUrl", EMPTY_STRING);
+            } else {
+                dashUrl = EMPTY_STRING;
+            }
         }
+        return dashUrl;
     }
 
     @Nonnull
     @Override
     public String getHlsUrl() throws ParsingException {
         assertPageFetched();
 
-        if (desktopStreamingData != null) {
-            return desktopStreamingData.getString("hlsManifestUrl");
-        } else if (mobileStreamingData != null) {
-            return mobileStreamingData.getString("hlsManifestUrl");
-        } else {
-            return EMPTY_STRING;
+        if (hlsUrl == null) {
+            if (desktopStreamingData != null) {
+                hlsUrl = desktopStreamingData.getString("hlsManifestUrl", EMPTY_STRING);
+            } else if (mobileStreamingData != null) {
+                hlsUrl = mobileStreamingData.getString("hlsManifestUrl", EMPTY_STRING);
+            } else {
+                hlsUrl = EMPTY_STRING;
+            }
         }
+        return hlsUrl;
     }
 
     @Override
     public List<AudioStream> getAudioStreams() throws ExtractionException {
         assertPageFetched();
-        final List<AudioStream> audioStreams = new ArrayList<>();
-        final YoutubeThrottlingDecrypter throttlingDecrypter = new YoutubeThrottlingDecrypter(getId());
 
-        try {
-            for (final Map.Entry<String, ItagItem> entry : getItags(ADAPTIVE_FORMATS,
-                    ItagItem.ItagType.AUDIO).entrySet()) {
-                final ItagItem itag = entry.getValue();
-                String url = entry.getKey();
-                url = throttlingDecrypter.apply(url);
-
-                final AudioStream audioStream = new AudioStream(url, itag);
-                if (!Stream.containSimilarStream(audioStream, audioStreams)) {
-                    audioStreams.add(audioStream);
+        if (audioStreams.isEmpty()) {
```

Comment:
If you are trying to cache the streams to prevent extracting them multiple times, you should initialize `audioStreams` with `null`. Otherwise the extraction will be run again if there aren't any streams.